floodwatch/src/js/components/Generate.js

```diff
@@ -1,54 +1,95 @@
-import d3 from 'd3';
-// import _ from 'lodash';
+//@flow
+
+import type {Filter, FilterLogic} from './filtertypes'
+import type {PersonResponse, FilterRequestItem} from '../api/types';
+import type {VisibilityMap} from './Compare'
+
+import {getVisibilityMap, generateDifferenceSentence, createSentence} from './comparisontools';
+import DemographicKeys from '../../stubbed_data/demographic_keys.json';
 import React, { Component } from 'react';
 import { Row, Col } from 'react-bootstrap';
 import { Chart } from './Chart';
 import {render} from 'react-dom';
 import url from 'url';
+import _ from 'lodash';
 
-import {getVisibilityMap, generateDifferenceSentence, createSentence} from './comparisontools';
-import type {VisibilityMap} from './Compare'
 
+type StateType = {
+  filterA: any,
```

Comment:
`FilterA` and `FilterB` are known types - they're `FilterRequestItem`s. Ditto on `dataA` and `dataB` being known.